server/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -117,7 +118,7 @@ public ChecksumBlobStoreFormat(
     public T read(String repoName, BlobContainer blobContainer, String name, NamedXContentRegistry namedXContentRegistry)
         throws IOException {
         String blobName = blobName(name);
-        try (InputStream in = blobContainer.readBlob(blobName)) {
+        try (InputStream in = blobContainer.readBlob(OperationPurpose.SNAPSHOT, blobName)) {
```

Comment:
Hmm this purpose should come from the caller I think (can be a followup tho)

Reply:
Yeah a follow up is better, see also [above](https://github.com/elastic/elasticsearch/pull/99615#discussion_r1337088235).